crypto/secp256k1/secp256k1.go

```diff
@@ -0,0 +1,205 @@
+package secp256k1
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"crypto/subtle"
+	"fmt"
+
+	secp256k1 "github.com/btcsuite/btcd/btcec"
+	amino "github.com/tendermint/go-amino"
+	"github.com/tendermint/tendermint/crypto"
+	"github.com/tendermint/tendermint/libs/common"
+	"golang.org/x/crypto/ripemd160"
+)
+
+//-------------------------------------
+const (
+	Secp256k1PrivKeyAminoRoute   = "tendermint/PrivKeySecp256k1"
+	Secp256k1PubKeyAminoRoute    = "tendermint/PubKeySecp256k1"
+	Secp256k1SignatureAminoRoute = "tendermint/SignatureSecp256k1"
+)
+
+var cdc = amino.NewCodec()
+
+func init() {
+	// NOTE: It's important that there be no conflicts here,
+	// as that would change the canonical representations,
+	// and therefore change the address.
+	// TODO: Add feature to go-amino to ensure that there
+	// are no conflicts.
```

Comment:
This comment is added everywhere. Not sure it is necessary everywhere. Also there should be an open issue in amino instead for (Add feature to go-amino to ensure that there are no conflicts).

Reply:
Agreed. Do you want that done in this same refactor or? That came from moving the file